eva/optimizer/column_deriver.py

```diff
@@ -0,0 +1,152 @@
+from eva.expression.constant_value_expression import ConstantValueExpression
+from eva.expression.tuple_value_expression import TupleValueExpression
+from eva.planner.abstract_plan import AbstractPlan
+from eva.planner.seq_scan_plan import SeqScanPlan
+from eva.planner.lateral_join_plan import LateralJoinPlan
+from eva.planner.storage_plan import StoragePlan
+from eva.planner.function_scan_plan import FunctionScanPlan
+from eva.planner.project_plan import ProjectPlan
+from eva.utils.single_dispatch_method import singledispatchmethod
+from eva.optimizer.column_mapper import ColumnMapper
+
+
+class ColumnDeriver:
+    def __init__(self):
+        pass
+    
+    @singledispatchmethod
+    def scan_node_for_attributes(self, node:AbstractPlan, alias_prefix:str=None) -> dict:
```

Comment:
In the context of DB, the scan keyword makes it confusing. 

Can we rename it to maybe `derive_column_mapping`?

Reply:
done.